R/importSNV.TCGA.R

```diff
@@ -0,0 +1,268 @@
+# importSNV.TCGA.R
+
+#' Import MAF files from source and converts them to rSNV files.
+#'
+#' \code{importSNV.TCGA} Imports data from one or more files of MAF format, creates rSNV or adds to
+#' existing rSNV file, returns rSNV file.
+#'
+#' @section Validations: validations sections has 3 functions - function that checks MAF file extension,
+#' function that checks MAF file header, function that checks rSNV file header.
+#'
+#' @section Read MAF file: maf file is read by extracting require columns from maf file followed
+#' by checking if every item in the class, type and chr columns are from a known and required list of values.
+#' These columns as a data table are then added to an rSNV file.
+#'
+#' @section Write Log: log section writes file processing progress along with other errors
+#' that may occur.
+#'
+#' @param fMAF vector of MAF file names.
+#' @param rSNV rSNV file to merged new MAF data with.
+#' @param na.rm Remove all rows with missing or "NA" values and drop them from data frame.
+#' @param silent Controls whether output to console should be suppressed. FALSE
+#'   by default.
+#' @param writeLog Controls whether writing the result to the global logfile is
+#'   enabled. TRUE by default.
+#'
+#' @family
+#'
+#'   ## @seealso \code{\link{importM}} importM() is used to import ...
+#'
+#'   ## @examples ## \dontrun {
+#'   ## importSNV.TCGA(c("aMAFfile.maf","bMAFfile"), "sampleSNV", na.rm = FALSE) ## }
+#' @export
+
+importSNV.TCGA <- function(  fMAF,
+                             rSNV,
+                             na.rm = FALSE,
+                             silent = FALSE,
+                             writeLog = TRUE) {
+
+    myNotes <- character()
+    myCall <- character()
+
+    # ==== PARAMETERS ==========================================================
+
+    # NL <- .PlatformLineBreak()
+    # # Read header and one contents line
+    # tmp <- readLines(fName, n = 2)
+
+    # ==== VALIDATIONS =========================================================
+
+    # General parameter checks
+    cR <- character()
+    cR <- c(cR, .checkArgs(fMAF,         like = rep("FILE_E", length(fMAF)),
+                                                             checkSize = TRUE))
```

Comment:
Check whether rSNV has a path prefix with the function ```dirname()``` which should be an existing directory.